pkg/engine/armresources.go

```diff
@@ -27,24 +27,30 @@ func GenerateARMResources(cs *api.ContainerService) []interface{} {
 		}
 	}
 
-	var useManagedIdentity, userAssignedIDEnabled bool
+	var useManagedIdentity, userAssignedIDEnabled, createNewUserAssignedIdentity bool
 	kubernetesConfig := cs.Properties.OrchestratorProfile.KubernetesConfig
 
 	if kubernetesConfig != nil {
 		useManagedIdentity = kubernetesConfig.UseManagedIdentity
-		userAssignedIDEnabled = useManagedIdentity && kubernetesConfig.UserAssignedID != ""
+		userAssignedIDEnabled = kubernetesConfig.UserAssignedIDEnabled()
+		createNewUserAssignedIdentity = kubernetesConfig.ShouldCreateNewUserAssignedIdentity()
 	}
 
 	isHostedMaster := cs.Properties.IsHostedMasterProfile()
 	if userAssignedIDEnabled {
-		userAssignedID := createUserAssignedIdentities()
+		if createNewUserAssignedIdentity {
```

Comment:
I'm curious, in which case do we need to create a new identity when there is already one asssigned by user?